src/iris/api.py

```diff
@@ -2775,6 +2784,39 @@ def on_put(self, req, resp, app_name):
                     logger.info('User %s has changed owners for app %s to: %s',
                                 req.context['username'], app_name, ', '.join(new_owners))
 
+            # change the sender address for the application
+            new_addresses = data.get('custom_sender_addresses')
+            if new_addresses:
+                # currently self service custom addresses are only implemented for the iris_smtp vendor but this can easily be extended
+                supported_custom_address_modes = ['email']
+                kill_address_modes = []
+
+                if not isinstance(new_addresses, dict):
+                    raise HTTPBadRequest('To change custom addresses, you must pass a dictionary of mode: address pairings')
+                for mode in new_addresses:
+                    if mode not in supported_custom_address_modes:
+                        raise HTTPBadRequest('%s does not support custom sender addresses', mode)
+
+                for mode in supported_custom_address_modes:
+                    if mode not in new_addresses:
```

Comment:
I would kill the address only if the caller specifies `email: None`. Otherwise, the caller needs to find out and then duplicate all the existing settings that they don't want to change.

Reply:
I don't have strong feelings about it either way so I'll change it to the behaviour you mentioned.